src/main/java/mate/academy/bookstore/mapper/OrderMapper.java

```diff
@@ -0,0 +1,38 @@
+package mate.academy.bookstore.mapper;
+
+import static mate.academy.bookstore.model.Order.Status.PENDING;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Set;
+import java.util.stream.Collectors;
+import mate.academy.bookstore.config.MapperConfig;
+import mate.academy.bookstore.dto.order.CreateOrderRequestDto;
+import mate.academy.bookstore.dto.order.OrderResponseDto;
+import mate.academy.bookstore.model.Order;
+import mate.academy.bookstore.model.OrderItem;
+import mate.academy.bookstore.model.User;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(config = MapperConfig.class, uses = OrderItemMapper.class)
+public interface OrderMapper {
+    @Mapping(target = "userId", source = "user.id")
+    OrderResponseDto toDto(Order order);
+
+    default Order toModel(CreateOrderRequestDto requestDto, User user, Set<OrderItem> orderItems) {
+        Order order = new Order();
+        order.setUser(user);
+        order.setOrderDate(LocalDateTime.now());
+        order.setStatus(PENDING);
+        order.setShippingAddress(requestDto.getShippingAddress());
+        order.setTotalPrice(orderItems.stream()
+                .map(item -> item.getPrice()
+                        .multiply(BigDecimal.valueOf(item.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add));
+        order.setOrderItems(orderItems.stream()
+                .peek(item -> item.setOrder(order))
+                .collect(Collectors.toSet()));
+        return order;
```

Comment:
not sure that this method fits into the mapper class, because we have a default status setting here, and calculate total price, which is more of a business logic, so let's move this to the service, but maybe keep it in a private method there, just name it properly